src/fft.rg

```diff
@@ -144,133 +152,435 @@ function fft.generate_fft_interface(itype, dtype)
+
+  --Task: Make plan in GPU version. Calls cufftPlanMany and stores plan in cufft_p
   local make_plan_gpu
-  if use_cuda then
+  if gpu_available then
     __demand(__cuda, __leaf)
-    task make_plan_gpu(input : region(ispace(itype), dtype),
-                       output : region(ispace(itype), dtype),
-                       address_space : c.legion_address_space_t) : cufft_c.cufftHandle
-    where reads writes(input, output) do
+    task make_plan_gpu(input : region(ispace(itype), dtype_in), output : region(ispace(itype), dtype_out), plan : region(ispace(int1d), iface.plan), address_space : c.legion_address_space_t)
+    
+    where reads writes(input, output, plan) do
+      format.println("In iface.make_plan_gpu...")
+
+      --Get pointer to plan
+      var p = iface.get_plan(plan, true)
+
+      --Verify we are in GPU mode by checking TOC_PROC
+      --Takes a c.legion_runtime_t and returns c.legion_runtime_get_executing_processor(runtime, ctx)
       var proc = get_executing_processor(__runtime())
+      format.println("Make_Plan_GPU: TOC PROC IS {}",c.TOC_PROC)
+      format.println("Make_Plan_GPU: Processor kind is {}", c.legion_processor_kind(proc))
+
       if c.legion_processor_kind(proc) == c.TOC_PROC then
+        format.println("Processor is TOC, so running GPU functions")
         var i = c.legion_processor_address_space(proc)
         regentlib.assert(address_space == i, "make_plan_gpu must be executed on a processor in the same address space")
 
-        var input_base = get_base(rect_t(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
-        var output_base = get_base(rect_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
+        --Get input and output bases
+        var input_base = get_base_in(rect_t_in(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
+        var output_base = get_base_out(rect_t_out(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
         var lo = input.ispace.bounds.lo:to_point()
         var hi = input.ispace.bounds.hi:to_point()
-        var n : int[dim]
+        var n : int[dim] --n is an array of size dim with the size of each dimension in the entries
         ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
-        var cufft_p : cufft_c.cufftHandle
-        var ok = cufft_c.cufftPlanMany(
-          &cufft_p,
-          dim,
-          &n[0],
-          [&int](0), 0, 0, -- inembed, istride, idist
-          [&int](0), 0, 0, -- onembed, ostride, odist
-          cufft_c.CUFFT_C2C,
-          1 -- batch
-        )
+      
+        --Call cufftPlanMany: cufftResult cufftPlanMany(cufftHandle *plan, int rank, int *n, int *inembed, int istride, int idist, int *onembed, int ostride, int odist, cufftType type, int batch) --rank = dimensionality of transform (1,2,3)
+        format.println("Calling cufftPlanMany...")
+
+        var ok = 0
```

Comment:
Instead of having an `ok` variable, I would make a `check_cufft` function or macro that checks the result, so you can write something like:

```
check_cufft(cufft_c.cufftPlanMany(...))
```

You can then write `check_cufft` to give you nice, readable error messages with string representations of the error codes and so on.

Reply:
[On my to-do list]